src/main/java/WayofTime/bloodmagic/command/sub/SubCommandNetwork.java

```diff
@@ -2,182 +2,215 @@
 import net.minecraft.util.text.TextComponentString;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraftforge.server.command.CommandTreeBase;
+import net.minecraftforge.server.command.CommandTreeHelp;
+
+import java.util.List;
+
+public class SubCommandNetwork extends CommandTreeBase {
+
+    public SubCommandNetwork() {
+        addSubcommand(new Syphon());
+        addSubcommand(new Add());
+        addSubcommand(new Set());
+        addSubcommand(new Get());
+        addSubcommand(new Cap());
+        addSubcommand(new Fill());
+        addSubcommand(new Tickets());
+        addSubcommand(new CommandTreeHelp(this));
+    }
 
-import java.util.Locale;
-
-public class SubCommandNetwork extends CommandBase {
     @Override
     public String getName() {
         return "network";
     }
 
     @Override
     public String getUsage(ICommandSender commandSender) {
-        return TextHelper.localizeEffect("commands.bloodmagic.network.usage");
+        return new TextComponentTranslation("commands.bloodmagic.network.usage").getFormattedText();
     }
 
     @Override
     public int getRequiredPermissionLevel() {
-        return 2;
+        return 0;
     }
 
-    @Override
-    public void execute(MinecraftServer server, ICommandSender commandSender, String[] args) throws CommandException {
-        if (args.length > 1) {
-            if (args[0].equalsIgnoreCase("help"))
-                return;
+    abstract class NetworkCommand extends CommandTreeBase {
+        public String help = new TextComponentTranslation("commands.bloodmagic.network." + getName() + ".help", getInfo()).getFormattedText();
+
+        public EntityPlayerMP player;
+        public SoulNetwork network;
+        public String uuid;
+
+        public Integer commandHelperAmount(MinecraftServer server, ICommandSender sender, String[] args) {
+            int amount;
+            if (args.length == 0)
+                amount = 1000;
+            else if (Utils.isInteger(args[0]))
+                amount = Integer.parseInt(args[0]);
+            else if (args.length > 1 && Utils.isInteger(args[1]))
+                amount = Integer.parseInt(args[1]);
+            else {
+                sender.sendMessage(new TextComponentTranslation("commands.bloodmagic.error.arg.invalid"));
+                sender.sendMessage(new TextComponentTranslation(this.getUsage(sender)));
+                return null;
+            }
+            if (amount < 0) {
+                sender.sendMessage(new TextComponentTranslation("commands.bloodmagic.error.negative"));
+                return null;
+            }
+            return amount;
+        }
 
-            try {
-                EntityPlayer player = CommandBase.getPlayer(server, commandSender, args[1]);
+        @Override
+        public String getUsage(ICommandSender sender) {
+            return new TextComponentTranslation("commands.bloodmagic.network." + getName() + ".usage").getFormattedText() + "\n" + help;
```

Comment:
`getUsage(...)` expects a localization key.

Reply:
It returns localization keys and they're getting translated.

Reply:
You are returning a localized value. It expects a ***key***.